httpd/handler.go

```diff
@@ -175,6 +179,127 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *influ
 	httpResults(w, results, pretty)
 }
 
+type Point struct {
+	Name      string                 `json:"name"`
+	Timestamp time.Time              `json:"timestamp"`
+	Tags      map[string]string      `json:"tags"`
+	Fields    map[string]interface{} `json:"fields"`
+}
+
+type Batch struct {
+	Database        string  `json:"database"`
+	RetentionPolicy string  `json:"retentionPolicy"`
+	Points          []Point `json:"points"`
+}
+
+func interfaceToString(v interface{}) string {
+	switch t := v.(type) {
+	case nil:
+		return ""
+	case bool:
+		return fmt.Sprintf("%v", v)
+	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, uintptr:
+		return fmt.Sprintf("%d", t)
+	case float32, float64:
+		return fmt.Sprintf("%v", t)
+	default:
+		return fmt.Sprintf("%v", t)
+	}
+}
+
+// Return all the measurements from the given DB
+func (h *Handler) showMeasurements(db string, user *influxdb.User) (error, []string) {
+	var measurements []string
+	queryString := "show measurements"
+	p := influxql.NewParser(strings.NewReader(queryString))
+	query, err := p.ParseQuery()
+	if err != nil {
+		return err, measurements
+	}
+	results := h.server.ExecuteQuery(query, db, user)
+	if results.Err != nil {
+		return results.Err, measurements
+
+	}
+	for _, measurementResult := range results.Results {
+		for _, measurementRow := range measurementResult.Series {
+			for _, measurementTuple := range (*measurementRow).Values {
+				for _, measurementCell := range measurementTuple {
+					measurements = append(measurements, interfaceToString(measurementCell))
+				}
+			}
+		}
+	}
+	return nil, measurements
+}
+
+// serveDump returns all points in the given database as a plaintext list of JSON structs.
+// To get all points:
+// Find all measurements (show measurements).
+// For each measurement do select * from <measurement> group by *
+func (h *Handler) serveDump(w http.ResponseWriter, r *http.Request, user *influxdb.User) {
+	q := r.URL.Query()
+	db := q.Get("db")
+	delim := []byte("\n")
+	pretty := q.Get("pretty") == "true"
+	err, measurements := h.showMeasurements(db, user)
+	if err != nil {
+		httpError(w, "error with dump: "+err.Error(), pretty, http.StatusBadRequest)
```

Comment:
I think this should be a `http. StatusInternalServerError ` as something went wrong on our end, not theirs.